### PR TITLE
[PX-2254] TCF v2.2 Enforcement Upon Parsing

### PR DESCRIPTION
**Pull request by kylehalverson**
Also updated README to reflect most current support and examples of the Library.

[JIRA Ticket](https://liveramp.atlassian.net/browse/PX-2254)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,3 +59,66 @@ func main() {
 ```
 
 The function `Parse(s string)` is deprecated, and should no longer be used.
+
+# Global Privacy Platform v1.0
+
```
Comment (gowthamkommineni) [inline README.md:64]: Drop a link to the actual GPP spec, if someone wants to read though!

File: parse.go [hunk 2]
```diff
@@ -427,6 +427,17 @@ func ParseV2(s string) (*V2ParsedConsent, error) {
 	p.SpecialFeaturesOptIn, _ = r.ReadBitField(12)
 	p.PurposesConsent, _ = r.ReadBitField(24)
 	p.PurposesLITransparency, _ = r.ReadBitField(24)
+	// Check for specific 2.2 Requirements and exit early.
+	// From IAB Docs: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#the-core-string
+	// "With TCF v2.2 support for legitimate interest for purpose 3 to 6 has been deprecated. Bits 2 to 5 are required to be set to 0."
```
Comment (gowthamkommineni) [inline parse.go:432]: , after v2.2

File: parse.go [hunk 3]
```diff
@@ -427,6 +427,17 @@ func ParseV2(s string) (*V2ParsedConsent, error) {
...
+	if mv, _ := p.MinorVersion(); mv >= 2 {
+		// Bitfield already adjusts to 1-indexing
```
Comment (gowthamkommineni) [inline parse.go]: Can you elaborate this comment please?!

**Review by gowthamkommineni — CHANGES_REQUESTED** [review]

File: parse.go (hunk 2 above)
Comment (kylehalverson) [reply parse.go:432]: This is a direct quote, so I added another line to help clarify.

File: parse.go (hunk 3 above)
Comment (kylehalverson) [reply parse.go]: Attempted to! Please let me know if that makes sense.

**Review by imang1 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (kylehalverson) [reply README.md:64]: Added.

**Review by gowthamkommineni — APPROVED** [review]

**Pull request merged**